feat(Accordion): made scrollable content focusable

thatblindgeye opened the pull request:
<!-- What changes are being made? Please link the issue being addressed. -->
**What**: Closes #8276 

[Accordion fixed with multiple expand example](https://patternfly-react-pr-8515.surge.sh/components/accordion#fixed-with-multiple-expand-behavior)

Also added an aria-labelledby prop and updated the fixed with multiple expand to show the scrollable bar.

<!-- Are there any upstream issues or separate issues you need to reference? -->
**Additional issues**:

patternfly-build commented:
Preview: https://patternfly-react-pr-8515.surge.sh

nicolethoen commented on packages/react-core/src/components/Accordion/AccordionContent.tsx:
```diff
@@ -31,29 +33,47 @@ export const AccordionContent: React.FunctionComponent<AccordionContentProps> =
+              className
+            )}
+            hidden={isHidden}
+            aria-label={ariaLabel}
```
I'm noticing a blank aria-label in the example, could the aria-label and aria-labelledby only be added as attributes if the props have been defined?

thatblindgeye replied:
Updated how those props are rendered. Considering the default value of aria-label is an empty string, I could also just remove its default value if not passed in.

nicolethoen reviewed (changes requested):
I am noticing a new github linting warning with your code changes
```
React Hook React.useEffect has a missing dependency: 'isFixed'. Either include it or remove the dependency array
```
That seems to indicate there are some best pracatices being violated. Is there a way we can avoid the warnings?

thatblindgeye commented:
>That seems to indicate there are some best pracatices being violated. Is there a way we can avoid the warnings?

@nicolethoen I added it to the dependency array and also added another conditional in the useEffect that should cover any sort of use case where `isFixed` might be conditionally updated.

nicolethoen reviewed (commented):
could you add a unit test to make sure this doesn't end up regressing in the future?

thatblindgeye commented:
@nicolethoen Added a test for passing in an arialabelledby for now. Testing that the content is actually scrollable/has a tab index and/or region seems a bit trickier as a unit test. @wise-king-sullyman do you have any thoughts on this?

thatblindgeye commented:
@nicolethoen @wise-king-sullyman Took a shot at adding an integration test, but it also may not be perfect since I don't believe there's any sort of way to simulate pressing the Tab key to move focus around in Cypress.

I added a couple of ways to somewhat test focusability at least, checking for tabindex and whether the element has focus after clicking it. Could also maybe try utilizing `scrollIntoView` in Cypress to sort of test scrollability, but I haven't tried that out yet.

nicolethoen reviewed (approved):
Testing that the content has a tabindex in this specific situation is the important thing. I think this LGTM!

wise-king-sullyman reviewed (approved):
This seems like a good approach to me as well!

kmcfaul reviewed (approved):
lgtm

mcoker commented on packages/react-core/src/components/Accordion/AccordionContent.tsx:
```diff
@@ -31,29 +33,49 @@ export const AccordionContent: React.FunctionComponent<AccordionContentProps> =
+            {...(ariaLabel && { 'aria-label': ariaLabel })}
+            {...(ariaLabelledby && { 'aria-labelledby': ariaLabelledby })}
+            {...(hasScrollbar && { tabIndex: 0 })}
+            {...(hasScrollbar && Container === 'div' && { role: 'region' })}
```
I don't know if it matters, but this will be false with `component="DIV"`.

mcoker commented on packages/react-core/src/components/Accordion/AccordionContent.tsx:
```diff
@@ -31,29 +33,49 @@ export const AccordionContent: React.FunctionComponent<AccordionContentProps> =
+              className
+            )}
+            hidden={isHidden}
+            {...(ariaLabel && { 'aria-label': ariaLabel })}
+            {...(ariaLabelledby && { 'aria-labelledby': ariaLabelledby })}
```
I'm sure it's fine, but in the example, we only add an `aria-labelledby` for the content element we know will scroll. Since there is a requirement for an accessible label when there is a scrollbar, I wonder if:
* we should add one if one is not currently set
* this may be possible already, but allow the user to access `hasScrollbar` and dynamically set one - I can see this as useful for accordions with dynamic or unknown content, or that go from scrollable to non for some other reason. In our example, if you change the content and the scrollbar isn't visible, one `AccordionContent` has a label and the rest do not. Also I noticed the changes that check `hasScrollbar` only seem to work on render, not when the size of the element changes and triggers a scrollbar to show/hide (like on page resize, toggle the nav open/closed, etc) - should it update?

mcoker reviewed (approved):
LGTM! A couple of comments but they aren't blockers.
